junifer/preprocess/warping/_fsl_warper.py

```diff
@@ -59,13 +59,28 @@ def preprocess(
             values and new ``reference_path`` key whose value points to the
             reference file used for warping.
 
+        Raises
+        ------
+        RuntimeError
+            If warp file path could not be found in ``extra_input``.
+
         """
         logger.debug("Using FSL for space warping")
 
         # Get the min of the voxel sizes from input and use it as the
         # resolution
         resolution = np.min(input["data"].header.get_zooms()[:3])
 
+        # Get warp file path
+        warp_file_path = None
+        for entry in extra_input["Warp"]:
+            if entry["dst"] == "native":
```

Comment:
Based on this logic and the `get_native_warper` function I asume that it's not possible for a datagrabber to provide transforms for the same space pair using two different warpers. Is this assumption correct?

Reply:
If so, can we add the check in the "validate" section of the datagrabber to make it more explicit?

Reply:
> Based on this logic and the `get_native_warper` function I asume that it's not possible for a datagrabber to provide transforms for the same space pair using two different warpers. Is this assumption correct?

Yes that's why we raise an error now just as we discussed. We can update the logic to take care of that when we actually have the transform files for two different warpers.

Reply:
> If so, can we add the check in the "validate" section of the datagrabber to make it more explicit?

Not sure I get you, can you please explain?

Reply:
I got mixed up. We don't really have a "validator" for post-datagrabbing consistency.

What I wanted to prevent is that someone creates a datagrabber which gives multiple warping options... not for now.

Reply:
Resolving then.